jira/client.py

```diff
@@ -259,6 +259,10 @@ def __init__(self, server=None, options=None, basic_auth=None, oauth=None, jwt=N
 
         self._options = copy.copy(JIRA.DEFAULT_OPTIONS)
 
+        if options.get('headers'):
+            self._options.get('headers').update(options.get('headers', {}))
```

Comment:
Nit: we already know that options.get('headers') is not `None` or empty if we make it into that branch, so you probably don't need the second `{}`...